### PR TITLE
feat(ingest): MongoDB ingestion source

### LINKED ISSUE #2280: Implement a connector for MongoDB

### PR DESCRIPTION
**Pull request by hsheth2**
Adds a MongoDB connector (basic list of databases and collections only) and associated integration test. Closes #2280.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                dataset_snapshot = DatasetSnapshot()
+                dataset_snapshot.urn = f"urn:li:dataset:(urn:li:dataPlatform:{platform},{dataset_name},{env})"
+
+                dataset_properties = DatasetPropertiesClass(
+                    tags=[],
+                    customProperties={},
+                )
+                dataset_snapshot.aspects.append(dataset_properties)
+
+                # TODO: Guess the schema via sampling
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/mongodb.py:105]: Would be great if mongo provided this type of capability for us.. ie tracking the superset document

**Review by jjoyce0510 — COMMENTED** [review]
awesome pr! is there any other metadata we can extract from mongo? comments, description etc?

**Review by shirshanka — APPROVED** [review]
LGTM!

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/mongodb.py:105]: There's nothing for it in mongodb itself, although the MongoDB Compass app does have this, albeit written in JS

**Comment by hsheth2** [comment]
> awesome pr! is there any other metadata we can extract from mongo? comments, description etc?

I don't think MongoDB has comments or descriptions 🙁

**Pull request merged**